### DISCUSSION TITLE
jaeger exporter

### DISCUSSION THREAD

**Discussion by shivchelwa** [open-telemetry/opentelemetry-java-instrumentation#1978] [Q&A] [answered]
Hi all, I am trying to enable auto instrumentation on my app and export the traces to jeager. As per the doc https://opentelemetry.io/docs/java/getting_started or https://github.com/open-telemetry/opentelemetry-java-instrumentation I should add below properties to start my app, but it does not tell where to find exporter for jeager. Do I need to build the jar from source?

java -javaagent:path/to/opentelemetry-javaagent-all.jar
-Dotel.exporter=jeager
-Dotel.exporter.jar=path/to/external-exporter.jar
-jar myapp.jar

**Comment by anuraaga** [comment, marked as answer]
Hi @shivchelwa - `jaeger` is included in the agent by default, so you don't need to specify anything for `otel.exporter.jar`. Just set `-Dotel.exporter=jaeger` and most likely the endpoint as mentioned here 

https://github.com/open-telemetry/opentelemetry-java-instrumentation#jaeger-exporter

**Reply by shivchelwa** [reply]
@anuraaga, thats what I tried first, but get below warning and the trace is not visible in Jaeger. 

`[opentelemetry.auto.trace 2021-01-04 22:35:34:771 -0500] [main] WARN io.opentelemetry.javaagent.tooling.OpenTelemetryInstaller - No jeager span exporter found`

I found below artifact on maven central, added it to project dependencies that didnt work. Hence downloaded the jar and set it to system property otel.exporter.jar=/path/to/jar. Now I don't see the warning and I can see trace in Jaeger.


<dependency>
    <groupId>io.opentelemetry.javaagent</groupId>
    <artifactId>opentelemetry-javaagent-exporter-jaeger</artifactId>
    <version>0.13.1</version>
</dependency>

Looks like this system property is must to use jaeger exporter. It will be useful if docs are updated to includes steps to find the jar or build from source.

**Reply by anuraaga** [reply]
Oh - it looks like you just have a typo. It should be `-Dotel.exporter=jaeger` (not _jeager_). Then it should work fine without the exporter.jar.
